### PR TITLE
[DPE-3202] Architecture specific snap revision

### PR DESCRIPTION
**Pull request by dragomirp**
Adds CPU architecture specific snap revisions. We will need this to keep on holding snap revisions on multiple architectures.

### CODE REVIEW

**Review by taurus-forever — CHANGES_REQUESTED** [review]
LGTM, but as discussed, please use `14/latest` snap track.

**Comment by dragomirp** [comment]
> LGTM, but as discussed, please use `14/latest` snap track.

`14/latest` or `14/stable`? Currently it's set to `14/stable`.

**Comment by taurus-forever** [comment]
> > LGTM, but as discussed, please use `14/latest` snap track.
> 
> `14/latest` or `14/stable`? Currently it's set to `14/stable`.

Midnight comment. sure `14/stable`. :-D

**Review by taurus-forever — APPROVED** [review]
tnx!

File: src/charm.py [hunk 1]
```diff
@@ -1274,9 +1272,18 @@ def _install_snap_packages(self, packages: List[str], refresh: bool = False) ->
                 snap_package = snap_cache[snap_name]
 
                 if not snap_package.present or refresh:
-                    if snap_version.get("revision"):
+                    if revision := snap_version.get("revision"):
+                        if type(revision) is dict:
```
[the remark this answers is not in the document]
Comment (marceloneppel) [reply src/charm.py:1276]: I think it's not needed, but also not a problem to keep it here.

**Review by marceloneppel — APPROVED** [review]
LGTM!

File: src/charm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dragomirp) [reply src/charm.py]: Removed for simplicity's sake.

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**